server/geoip_service.py

```diff
@@ -94,14 +117,42 @@ def __init__(self):
         """
 
         chunk_size = 1024
-        async with aiohttp.ClientSession() as session:
-            async with session.get(url, timeout=60 * 20) as resp:
+        params = {
+            "edition_id": "GeoLite2-Country",
+            "license_key": license_key,
+            "suffix": "tar.gz"
+        }
+
+        async def get_checksum(session):
+            async with session.get(url, params={
+                **params,
+                "suffix": f"{params['suffix']}.md5"
+            }, timeout=60 * 20) as resp:
+                return await resp.text()
+
+        async def get_db_file_with_checksum(session):
+            hasher = hashlib.md5()
+            async with session.get(url, params=params, timeout=60 * 20) as resp:
                 with open(file_path, 'wb') as f:
                     while True:
                         chunk = await resp.content.read(chunk_size)
                         if not chunk:
                             break
                         f.write(chunk)
+                        hasher.update(chunk)
+
+            return hasher.hexdigest()
+
+        async with aiohttp.ClientSession(raise_for_status=True) as session:
+            checksum, our_hash = await asyncio.gather(
+                get_checksum(session),
+                get_db_file_with_checksum(session)
+            )
+
+        if checksum != our_hash:
+            raise Exception(
+                f"Hashes did not match! Expected {checksum} got {our_hash}"
+            )
```

Comment:
shouldn't you log?

Reply:
It will get logged by the exception handler